README.md

```diff
@@ -1,13 +1,143 @@
-# The Aura Compiler
+<h1 align="center">
+    The Aura compiler (<code>aurac</code>) 
+</h1>
 
-`aurac` is a [Aura](https://github.com/auralangco/aura) compiler that targets `C` (and probably `JS` in the future).
+<div  align="center">
+    <img src="https://img.shields.io/github/stars/auralangco/aurac" alt="GitHub stars" href="https://github.com/auralangco/aurac/stargazers" />
+    <img src="https://img.shields.io/github/release/auralangco/aurac" alt="GitHub release" href="https://github.com/auralangco/aurac/releases" />
+    <img src="https://img.shields.io/github/license/auralangco/aurac" alt="License" href="https://raw.githubusercontent.com/auralangco/aurac/master/LICENSE" />
+</div>
 
-Currently, both `aurac` and the Aura Lang are on pre-alpha versions.
+<p align="center" style="text-align: justify;">
+    <code>aurac</code> is the official compiler for the <strong>Aura programming language</strong>, a functional, parallelizable, and multitarget language. The current version targets the C programming language, with plans to extend support to JavaScript and other languages in the future.
+</p>
+
+## Table of Contents
+
+- [Features](#features)
+- [Getting Started](#getting-started)
+  - [Prerequisites](#prerequisites)
+  - [Installation](#installation)
+- [Usage](#usage)
+- [Example Code](#example-code)
+- [Future Plans](#future-plans)
+- [Contributing](#contributing)
+- [License](#license)
+- [Contact](#contact)
+
+## Features
+
+- **Functional Language**: Emphasizes immutability and functional programming paradigms.
+- **Parallelizable**: Designed to run efficiently on multicore processors.
+- **Multitarget**: Initially targets C, with plans for JavaScript and more.
+- **Type Safety**: Ensures operations are validated during compile time.
+- **Compile-Time and Runtime Operations**: Allows compile-time operations to be available at runtime while enforcing proper compile-time treatment.
+
+## Getting Started
+
+To get started with `aurac`, clone the repository and follow the instructions below to set up and use the compiler.
+
+### Prerequisites
+
+- [Rust](https://www.rust-lang.org/tools/install) (required to build the compiler)
+
+- [Cargo](https://doc.rust-lang.org/cargo/getting-started/installation.html) (required to build the compiler)
+
+- A `C` compiler (required to compile the generated C code)
+
+### Installation
+
+1. Clone the repository:
+
+   ```sh
+   git clone https://github.com/auralangco/aurac.git
+   cd aurac
+   ```
+
+2. Build the project:
+   ```sh
+   cargo build
+   ```
+
+## Usage
+
+To compile an Aura program, use the following command:
+
+```sh
+cargo run -- path/to/yourfile.aura
+```
```

Comment:
At the moment the front-end isn't implemented, so compiling aura source code isn't yet supported the user writes the IR directly. Don't think it's worth mentioning it at the moment